src/Infrastructure/ViewRulesCompiler.js

```diff
@@ -0,0 +1,60 @@
+import IfcColor from './IfcColor'
+import IfcCustomViewSettings from './IfcCustomViewSettings'
+import {IFCPROPERTYSET, IFCRELDEFINESBYPROPERTIES} from 'web-ifc'
+import {interpolateColors, parseColor} from './ColorHelperFunctions'
+
+
+/* eslint-disable new-cap */
+/* eslint-disable no-magic-numbers */
+/**
+ * Convert view rules into view settings object
+ *
+ * @param {object} api
+ * @param {number} modelID
+ * @param {number} rules
+ * @return {IfcCustomViewSettings} the view settings
+ */
+export async function compileViewRules(api, modelID, rules) {
```

Comment:
Please add unit tests

Reply:
This is currently just a placeholder for the rule compilation functionality